skrifa/src/attribute.rs

```diff
@@ -0,0 +1,256 @@
+//! Primary attributes typically used for font classification and selection.
+
+use read_fonts::{types::Tag, TableProvider};
+
+/// Stretch, style and weight attributes of a font.
+///
+/// Variable fonts may contain axes that modify these attributes. The
+/// [new](Self::new) method on this type returns values for the default
+/// instance.
+///
+/// These are derived from values in the
+/// [OS/2](https://learn.microsoft.com/en-us/typography/opentype/spec/os2) if
+/// available. Otherwise, they are retrieved from the
+/// [head](https://learn.microsoft.com/en-us/typography/opentype/spec/head)
+/// table.
+#[derive(Copy, Clone, PartialEq, Debug, Default)]
+pub struct Attributes {
+    pub stretch: Stretch,
+    pub style: Style,
+    pub weight: Weight,
+}
+
+impl Attributes {
+    /// Extracts the stretch, style and weight attributes for the default
+    /// instance of the given font.
+    pub fn new<'a>(font: &impl TableProvider<'a>) -> Self {
+        let mut stretch = Stretch::default();
+        let mut style = Style::default();
+        let mut weight = Weight::default();
```

Comment:
Total nit/preference thing but in general I try to avoid the pattern of declaring mutable variables and then setting them unless I absolutely have to, since it makes the program flow harder to follow. I think that personally, in this case, I would probably have separate helper methods for the `os2` and `head` branches, with each just returning an `Attributes` struct 🤷

Reply:
Seems reasonable! I’ll break this up and use control flow rather than mutable variables.

Reply:
Done. This looks much cleaner, ty.